pkg/csi_driver/node.go

```diff
@@ -148,6 +148,9 @@ func (s *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublish
 		}
 		klog.V(6).Infof("NodePublishVolume populating identity provider %q in mount options", identityProvider)
 		fuseMountOptions = joinMountOptions(fuseMountOptions, []string{util.OptInHnw + "=true", util.TokenServerIdentityProviderConst + "=" + identityProvider})
+	} else if enableSidecarBucketAccessCheckForSidecarVersion {
```

Comment:
Is there an easy way you can add unit tests for this? I see we didn't add unit tests for any of the sidecar access check stuff / checking that correct fuse mount Options are passed. Can you add a test case for this (hostnetwork + enableSidecarBucketAccessCheckForSidecarVersion=true/false), not hostnetwork + enableSidecarBucketAccessCheckForSidecarVersion=true/false. At first glance, it looks pretty straight forward: https://github.com/GoogleCloudPlatform/gcs-fuse-csi-driver/blob/9e0ddcb4a6d9eff48967744990cd1dd45266abb9/pkg/csi_driver/node_test.go#L134-L149. If it turns out to be too hard to implement, then you can ignore, but seems like it will be an easy test case to add.